File: Model/Person/Tourist.cs
public class Tourist : Person
{
    private float budget;
    private float joyLevel;
    public Tourist(string name)
    {
        super(name);
        joyLevel = 100;
    }
    public void takePhoto(Animal animal)
    {

    }
    public void observeAnimals()
    {

    }
    public void increaseJoy(double amount)
    {
        if (joyLevel + amount <= 100)
        {

            joyLevel += joyLevel;
        }
    }
    public void decreaseJoy(double amount)
    {
        if (joyLevel - amount >= 0)
        {

            joyLevel -= joyLevel;
        }
    }
    public void spendMoney(double amount){
        budget -= amount;
    }
    //NO BUY ANIMAL ADDED BECAUSE ANIMALS CANT BE BOUGHT BY TOURISTS
    public double getJoyLevel(){
        return joyLevel;
    }
}

File: Model/Person/Tourist.java
package Model.Person;

import Model.Animal.Animal;

public class Tourist extends Person {
    private double budget;
    private double joyLevel;

    public Tourist(String name, PersonType type) {
        super(name, type);
        this.joyLevel = 100;
    }

    public void takePhoto(Animal animal) {
        // TODO: Implement photo taking logic
    }

    public void observeAnimals() {
        // TODO: Implement animal observation logic
    }

    public void increaseJoy(double amount) {
        if (joyLevel + amount <= 100) {
            joyLevel += amount;
        }
    }

    public void decreaseJoy(double amount) {
        if (joyLevel - amount >= 0) {
            joyLevel -= amount;
        }
    }

    public void spendMoney(double amount) {
        budget -= amount;
    }

    public double getJoyLevel() {
        return joyLevel;
    }
} 